### ISSUE TITLE
Items needed for v2.7 release

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#1212]
We plan on releasing v2.7 to beta in 1-2 weeks' time. :) Below are the items that need to be completed before the release:

- [x] #252 , up to the category suggestion stage
- [x] Memory leaks in Nearby solved #1368
- [x] Other Nearby fixes (more detail in #1329 ):
   - Add "?" description fix to NearbyList
   - Known crashes fixed
   - Test that uploads, including direct uploads (from list and map), non-direct uploads, and multiple share uploads work as expected
- [x] #7 , including changes requested
- [x] #686 if possible
- [x] #1337 merged since we don't want an incomplete feature
- [x] All of the above merged into master and tested on prodRelease build
- [x] Make sure that Dagger NPE fixes from previous release are included

**Comment by neslihanturan** [comment]
#686 comes tomorrow:)

**Comment by VojtechDostal** [comment]
I think we should start featuring these updates on [Tech News](https://meta.wikimedia.org/wiki/Tech/News#contribute), which are weekly sent to village pumps of many language Wikipedias and thus could provide a lot of publicity for us. The next issue is scheduled for publication on 2018-03-05, another on 2018-03-12. Questions:

1. should we announce the date the change comes to beta? or the final deployment? I think the latter is correct.
2. should we announce in advance or after the change? I think after is OK; we often don't know the date precisely.  

So it would be cool if someone tags me whenever a major change is released and I'll post it to Tech News.

**Comment by misaochan** [comment]
Thanks @VojtechDostal ! Will ping you when we do the release. 

1. I would go with the latter, but beta isn't too bad either if we include a link to register for beta IMO.
2. Definitely after. It's really hard (and stressful :)) to pinpoint an exact date beforehand. Plus people will likely just forget when the time arrives

**Comment by misaochan** [comment]
I created the release branch https://github.com/commons-app/apps-android-commons/tree/2.7.x-release and switched the base for the 2 pending PRs required for release. @neslihanturan please feel free to test the prodRelease build on that branch. :)

Aside from PRs needed to fix urgent issues with 2.7, all other PRs and new features will go into master and scheduled for 2.8 release.

**Comment by misaochan** [comment]
A huge thank you and congratulations to everyone who has contributed to the app so far! All of your contributions will be live in the Play Store version of the app soon. :) I plan on pushing the beta release tomorrow (since I will be heading to bed soon and won't be able to test the Play Store version if I do it today). 

@VojtechDostal what information will you need for the Tech News update? I can draft that and send it to you after the release is out. I also plan on writing a blog post that we can link on FB, but that might take a bit longer to do.

**Comment by neslihanturan** [comment]
Best release ever! Thanks everyone who contributed:)

**Comment by misaochan** [comment]
Wrote a summary of the new features at 
https://cookiesandcodeblog.wordpress.com/2018/04/21/commons-app-version-2-7-beta-release/ :)

Fingers crossed everything will be okay with 2.7.1 and we can push to production soon.
